lib/retry-delay.js

```diff
@@ -0,0 +1,55 @@
+// Encapsulation of configurable backoff/jitter behavior.
+//
+// - The system can either be in a "good" state or a "bad" state. The initial state is "bad"; the
+// caller is responsible for indicating when it transitions to "good". When we ask for a new retry
+// delay, that implies the state is now transitioning to "bad".
+//
+// - There is a configurable base delay, which can be changed at any time (if the SSE server sends
+// us a "retry:" directive).
+//
+// - There are optional strategies for applying backoff and jitter to the delay.
+
```

Comment:
This is basically a straight port of the corresponding logic that I wrote in the Go `eventsource`. It's not particularly elegant, but I think it's at least straightforward to read and to test.